### ISSUE TITLE
`OSError: [Errno 8] Bad file descriptor` when trying to save `.npy` files

### ISSUE THREAD

**Issue by agriyakhetarpal** [pyodide/pyodide#4779]
## 🐛 Bug

This bug report is quite similar to #4541 that was filed a while ago, except for the fact that this has a different error number (8) instead of (9), and also, this is more related to NumPy's behaviour of saving `.npy` files, rather than loading them when running in the Emscripten file system.

### To Reproduce

I am trying to build and test the `numcodecs` Python package out-of-tree with Pyodide which is what brings these errors, so I do not have a proper reproducer but something minimal like this _should_ fail:

```python
import numpy

arr = numpy.array([1, 2, 3, 4, 5])
numpy.save("arr.npy", arr)
```

### Expected behavior

Pyodide's runtime should be able to both save and load `.npy` files without issues.

### Environment

- Pyodide Version: `pyodide-build` version 0.25.1, Emscripten 3.1.46
- Browser version: N/A
- Any other relevant information: N/A

### Additional context

Previously, a [patch](https://github.com/hoodmane/pyodide/blob/0b94792c3934ba2a858ed15938a3127057fc4a01/emsdk/patches/0001-Fix-dup-in-nodefs-by-refcounting-nodefs-file-descrip.patch) has been used to fix the problem for loading the `.npy` files after https://github.com/emscripten-core/emscripten/pull/21399 was merged, which is first available in Emscripten 3.1.55, and therefore will be available in [Pyodide 0.26.0](https://github.com/pyodide/pyodide/issues/4713) (Emscripten 3.1.58). However, this patch does not fix the action of saving them to the file system.

**Comment by hoodmane** [comment]
Interesting. These are generally pretty fun bugs to work on. Highest probability is that there is yet another bug with dup() in the nodefs.

**Comment by agriyakhetarpal** [comment]
I'm not sure where or when this was fixed – I tried this again and I can't reproduce the error now; neither with a Python REPL running from a Pyodide virtual environment an xbuildenv running Pyodide 0.27.0a2, nor with the latest version of Pyodide from `main` from the web console, nor with a simple `pytest` invocation. Both saving and loading operations on arrays with `.npy` files work now. Thanks, and closing!
